spec/schemas/_common.query_dsl.yaml

```diff
@@ -2094,6 +2097,28 @@ components:
           required:
             - big
             - little
+    TemplateQuery:
+      type: object
+      properties:
+        knn:
+          $ref: '#/components/schemas/TemplateKnnQuery'
+    TemplateKnnQuery:
```

Comment:
I don't think we should add this `TemplateKnnQuery` , template query can support any opensearch query, knn is just one kind . https://docs.opensearch.org/docs/latest/search-plugins/search-relevance/template-query/